app/test_main.py

```diff
@@ -1 +1,31 @@
-# write your code here
+import pytest
+from unittest.mock import patch
+
+from app.main import (can_access_google_page)
+
+
+@pytest.mark.parametrize(
+    "url, has_internet, is_valid, expected_result",
+    [
+        ("http://www.google.com", True, True, "Accessible"),
+        ("", False, True, "Not accessible"),
+        ("http://www.google.com", True, False, "Not accessible"),
+        ("", False, False, "Not accessible"),
+    ]
+)
```

Comment:
You don't need to write these URLs here; you can just pass one URL in the can_access_google_page function. Since you already have mocked the value valid_url, it will work as expected.